libtenzir/builtins/operators/http.cpp

```diff
@@ -366,12 +427,44 @@ auto find_plugin_for_mime(std::string_view mime, location op,
 }
 
 auto make_pipeline(const std::optional<located<pipeline>>& pipe,
-                   const http::response& r, location oploc,
+                   const caf::uri& uri, const http::response& r, location oploc,
                    diagnostic_handler& dh, bool is_warning)
   -> failure_or<located<pipeline>> {
   if (pipe) {
     return *pipe;
   }
+  auto parsed = boost::urls::parse_uri_reference(uri.str());
+  if (not parsed) {
+    diagnostic::error("invalid URI `{}`", uri.str()).primary(oploc).emit(dh);
+    return failure::promise();
+  }
+  if (not parsed->segments().empty()) {
+    auto segment = parsed->segments().back();
+    if (const auto lpos = segment.rfind('.'); lpos != std::string::npos
+                                              and lpos + 1 != segment.size()
+                                              and lpos != 0) {
+      auto extension = segment.substr(lpos + 1);
+      auto v = std::vector<operator_ptr>{};
+      TENZIR_DEBUG("[http] finding decompression plugin for extension `{}`",
+                   extension);
+      TRY(auto decompressor, find_decompression_plugin(extension, oploc, dh));
+      if (decompressor) {
+        v.push_back(std::move(decompressor));
+        if (const auto fpos = segment.rfind('.', lpos - 1);
+            fpos != std::string::npos and fpos + 1 != segment.size()
+            and fpos != 0) {
```

Comment:
The `fpos + 1 != segment.size()` looks fishy. It's always true here. You probably wanted something with `lpos`. (By the way, no idea what these names stand for, doesn't help too much).

Reply:
I see. Code needs to handle more than 2 dots as well, but looks like it does.

Reply:
Yep, only cares about 2 dots at most from the end of the segment.